src/nix/install.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+#-------------------------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+#-------------------------------------------------------------------------------------------------------------
+#
+# Docs: https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/sshd.md
+# Maintainer: The VS Code and Codespaces Teams
+#
+
+set -e
+
+if [ "$(id -u)" -ne 0 ]; then
+    echo -e 'Script must be run as root. Use sudo, su, or add "USER root" to your Dockerfile before running this script.'
+    exit 1
+fi
+
+
+sh <(curl -L https://nixos.org/nix/install) --daemon
```

Comment:
I'm not very familiar with Nix. Do you know if it works well in daemon mode in docker containers? Do we need to add anything to the `entrypoint` to start the daemon on container startup?

Reply:
Yeah an enterpoint would be needed.  This I think would also require the container run as root - which isn't a problem for services like Codespaces or when using Docker Desktop, but is a consideration otherwise.

Most of the exploration I did for https://github.com/Chuxel/feature-library/blob/main/src/nix/install.sh was trying to figure out a way to use single user unless the UID/GID changed. I think I have a working model.  We could opt to start from here as a baseline if preferrable.

It also verifies the download using their GPG key, has support for referencing a list of packages, a flake, or a derivation.